### PR TITLE
Extracting generic return type's actual runtime type

### PR DESCRIPTION
**Pull request by martingy**
With this change, extracting the actual return type of the generic super class's methods generic return type became possible.
The java version is 1.6.

### CODE REVIEW

**Comment by fehguy** [comment]
@martingy thank you.  What affect does bringing `classmate` into the picture have on backwards compatibility with Jackson?

**Comment by martingy** [comment]
@fehguy no affect on that as far as I can work out. classmate is another library from fasterxml, it is independent of Jackson.

**Comment by fehguy** [comment]
@martingy could you please rebase?

File: modules/swagger-jaxrs/src/main/java/io/swagger/jaxrs/Reader.java [hunk 1]
```diff
@@ -823,7 +817,22 @@ private Operation parseMethod(Class<?> cls, Method method, List<Parameter> globa
...
+                    && resolvedMethod != null && resolvedMethod.getReturnType() != null
+                    && resolvedMethod.getReturnType() instanceof ResolvedObjectType) {
+                responseType = ((ResolvedObjectType) resolvedMethod.getReturnType()).getErasedType();
+            } else if (method.getGenericReturnType() != null && method.getGenericReturnType() instanceof GenericArrayType
+                    && resolvedMethod != null && resolvedMethod.getReturnType() != null
+                    && resolvedMethod.getReturnType() instanceof ResolvedArrayType) {
+                responseType = ((ResolvedArrayType) resolvedMethod.getReturnType()).getErasedType();
+            }
+
+            LOGGER.debug("response class of method " + method + " is " + responseType);
```
Comment (pjfanning) [inline modules/swagger-jaxrs/src/main/java/io/swagger/jaxrs/Reader.java]: Could you use`LOGGER.debug("response class of method {} is {}", method, responseType`? https://www.slf4j.org/faq.html#logging_performance

File: pom.xml [hunk 2]
```diff
@@ -512,5 +512,7 @@
         <commons-io-version>2.4</commons-io-version>
         <enforcer-plugin-version>1.4</enforcer-plugin-version>
         <failsafe-plugin-version>2.19.1</failsafe-plugin-version>
+
+        <java.version>1.8</java.version>
```
Comment (pjfanning) [inline pom.xml]: I'm not sure that removing support for JRE 1.6 and JRE 1.7 should be done in a minor release.

**Pull request closed without merging**